Source/Core/Common/TraversalClient.cpp

```diff
@@ -71,12 +76,16 @@ static ENetAddress MakeENetAddress(const TraversalInetAddress& address)
   ENetAddress eaddr{};
   if (address.isIPV6)
   {
-    eaddr.port = 0;  // no support yet :(
+    memcpy(&eaddr.host, &address.address, sizeof(address.address));
+    eaddr.port = ntohs(address.port);
+    eaddr.sin6_scope_id = 0;
   }
   else
   {
-    eaddr.host = address.address[0];
+    u32 ipv4_mapped_addr[4] = {0, 0, 0xffff0000, address.address[0]};
+    memcpy(&eaddr.host, &ipv4_mapped_addr, sizeof(ipv4_mapped_addr));
```

Comment:
I think this is the second place where you make one of these "ipv4 mapped" addresses.
Maybe this logic should be moved into a little helper function?